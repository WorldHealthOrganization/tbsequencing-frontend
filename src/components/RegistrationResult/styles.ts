import { css } from '@emotion/react';

export const container = css({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
});

export const header = css({
  textAlign: 'center',
  marginTop: 156,
  marginBottom: 176,
  maxWidth: 798,
});
