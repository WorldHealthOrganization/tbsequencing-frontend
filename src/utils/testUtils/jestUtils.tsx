import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
