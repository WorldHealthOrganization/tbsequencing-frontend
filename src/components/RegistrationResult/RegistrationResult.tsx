/** @jsxImportSource @emotion/react */
import React from 'react';
import * as styles from './styles';
import H1 from '../typography/H1';
import { ReactComponent as FooterImage } from './assets/Frame.svg';

export const RegistrationResult = () => (
  <div css={styles.container}>
    <H1 style={styles.header}>
      To finalize your registration,
      please check your inbox and your spam folder
      after a few minutes to confirm your email address.
    </H1>
    <FooterImage width={1113} height={512} />
  </div>
);

export default RegistrationResult;
