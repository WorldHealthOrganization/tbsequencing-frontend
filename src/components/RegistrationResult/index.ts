import RegistrationResult from './RegistrationResult';

export default RegistrationResult;
