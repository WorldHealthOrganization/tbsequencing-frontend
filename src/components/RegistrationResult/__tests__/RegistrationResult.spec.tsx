import React from 'react';
import { getTreeJSONWithRedux, render } from '../../../utils/testUtils/jestUtils';
import RegistrationResult from '../RegistrationResult';

describe('Registration result tests', () => {
  test('renders RegistrationResult component', () => {
    render(
      <RegistrationResult />,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <RegistrationResult />,
    );

    expect(tree).toMatchSnapshot();
  });
});
