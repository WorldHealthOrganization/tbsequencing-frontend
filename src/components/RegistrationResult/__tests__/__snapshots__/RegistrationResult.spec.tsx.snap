// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration result tests Should render correctly 1`] = `
<div
  className="css-1p4btuk-RegistrationResult"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-idpj3q-MuiTypography-root-BaseText"
  >
    To finalize your registration, please check your inbox and your spam folder after a few minutes to confirm your email address.
  </p>
  <svg
    height={512}
    width={1113}
  >
    Frame.svg
  </svg>
</div>
`;
